### ISSUE TITLE
Loops based on hasNextRow() will skip last row

### ISSUE THREAD

**Issue by tim-slifer** [tim-slifer/superfly-csv#1]
The boolean return of `hasNextRow()` is always correct. However, when looping over the `CsvFile` using `hasNextRow()` as the loop condition, the last row on the file will be skipped. If the file only has one row, the loop will not be run altogether.

This is rooted in the approach of mimicking an Iterator with the navigation, but trying to expand on it by allowing backwards navigation and direct row focus. It would have been better to have just wrapped an Iterator, or return one directly.

**Comment by tim-slifer** [comment]
It's a conceptual difference. My approach has a concept of "current" index, whereas an Iterator does not. Trying to replicate the behavior of an Iterator while including my approach is what caused problems. I'm looking into wrapping an Iterator of some sort to handle the navigation between rows.

If I only care about moving focus to one row to the next and so forth, a standard Iterator will work. I can maintain the two-way movement and even the jump-to-row functionality with a ListIterator. The rub is, without the concept of a "current" index, using the `.next()` and `.previous()` methods in alternating order will actually return the same row.

**Comment by tim-slifer** [comment]
This one took me down the rabbit hole.

I've got a fix for the issue, but the solution ended up being a fairly deep redesign of the CsvLoader and CsvFile. Those details will be in the release notes.

To address this issue, I've dropped the `.hasNextRow()` method completely. Trying to operate like an Iterator only partially did not yield good results. I also tried to wrap the iterator behind the CsvFile API only to experience continued issues.

In the end, I decided that in order to have a working iteration approach, as well as to retain all the basic functionality (navigating next and previous, jumping to a row, etc), I'd need to change how I loop over the rows.

The recommended approach for this is now an enhanced `for` loop:

```java
for (CsvRow row : csv.getRows()) {
    // add tasks to be completed for each currently stored row.
}
```

This approach will reliably loop over each row on the CSV, without any skips or unintended exceptions due to how indexes were being tracked. And, if I'm being honest, I prefer a `for` over a `while`.
